src/github.com/couchbase/sync_gateway/db/index_changes.go

```diff
@@ -198,18 +145,13 @@ func (db *Database) VectorMultiChangesFeed(chans base.Set, options ChangesOption
 					db.addDocToChangeEntry(minEntry, options)
 				}
 
+				// Clock handling
 				if minEntry.Seq.TriggeredBy == 0 {
 					// Update the cumulative clock, and stick it on the entry.
 					cumulativeClock.SetMaxSequence(minEntry.Seq.vbNo, minEntry.Seq.Seq)
-					clockHash, err := db.SequenceHasher.GetHash(cumulativeClock)
-					// Change entries only need the hash value, not the full clock.  Creating a new
-					// clock here to avoid the overhead of cumulativeClock.copy()
-					minEntry.Seq.Clock = &base.SequenceClockImpl{}
-					if err != nil {
-						base.Warn("Error calculating hash for clock:%v", base.PrintClock(minEntry.Seq.Clock))
-					} else {
-						minEntry.Seq.Clock.SetHashedValue(clockHash)
-					}
+					// TODO: Potential enhancement - for first iteration loop, always set hash on last entry.
+					continuousLastHash = db.calculateHashWhenNeeded(options, minEntry, cumulativeClock, &hashedEntryCount, continuousLastHash)
+
```

Comment:
I think this would be easier to read as:

db.calculateHashWhenNeeded(
                                  options, 
                                  minEntry, 
                                  cumulativeClock, 
                                  &hashedEntryCount, 
                                  continuousLastHash,
)

and highlight the complexity being done in `calculateHashWhenNeeded()`